### PR TITLE
chore(ui5-dialog): migrate tests to Cypress

### PR DESCRIPTION
**Pull request by kskondov**
Migrates tests to cypress
JIRA: BGSOFUIRODOPI-3462

### CODE REVIEW

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 1]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+			</>
+		);
+
+		cy.get("#dialog")
+			.should("not.be.visible");
+		cy.get("#dialog")
+			.invoke("prop", "open", true);
+		cy.get("#dialog")
+			.should("be.visible");
+		cy.get("#btnCloseDialog").click();
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: Directyly call `cy.get("#dialog").invoke("prop", "open", false);` and delete the close button

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 2]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+			</>
+		);
+
+		cy.get("#dlgAttr")
+			.should("not.be.visible");
+		cy.get("#dlgAttr")
+			.invoke("attr", "open", true);
+		cy.get("#dlgAttr")
+			.should("be.visible");
+		cy.get("#btnCloseWithMethod").realClick();
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: Directly call `cy.get("#dialog").invoke("attr", "open", false);` and delete the close button

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 3]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+				</Dialog>
+			</>
+		);
+
+		cy.viewport(1000, 800);
+
+		// Open dialog
+		cy.get("#dialog").then(function (dialog) {
+			dialog[0].setAttribute("open", "true")
+		})
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: `cy.get("#dialog").invoke("attr", "open", true);`

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 4]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+				const topBeforeScreenResize = parseInt(dialog.css("top"));
+				const leftBeforeScreenResize = parseInt(dialog.css("left"));
+
+				cy.get("#dialog-close").click();
+				// Resize viewport
+				cy.viewport(2000, 2000);
+
+				cy.get("#dialog").then(function (dialog) {
+					dialog[0].setAttribute("open", "true")
+				})
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: `cy.get("#dialog").invoke("attr", "open", true);`

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 5]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		})
+		cy.get("#dialog").should("be.visible");
+
+		// Capture position before resize
+		cy.get("#dialog")
+			.then(dialog => {
+				const topBeforeScreenResize = parseInt(dialog.css("top"));
+				const leftBeforeScreenResize = parseInt(dialog.css("left"));
+
+				cy.get("#dialog-close").click();
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: `cy.get("#dialog").invoke("attr", "open", false);` and delete the close button logic

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 6]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+						// Assert position changes
+						expect(topBeforeScreenResize).not.to.equal(topAfterScreenResize);
+						expect(leftBeforeScreenResize).not.to.equal(leftAfterScreenResize);
+					});
+			});
+
+		// Close dialog
+		cy.get("#dialog-close").click();
+		cy.get("#dialog").should("not.be.visible");
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: not needed, delete it

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 7]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+	});
+
+	it("draggable and resizable dialog repositions after screen resize", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:425]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 8]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+						expect(leftBeforeScreenResize).not.to.equal(leftAfterScreenResize);
+					});
+			});
+
+		// Close dialog
+		cy.get("#draggable-and-resizable-close").click();
+		cy.get("#draggable-and-resizable-dialog").should("not.be.visible");
+	});
+
+	it("draggable - mouse support", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:478]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 9]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+								// Close dialog
+								cy.get("#draggable-close").click();
+								cy.get("#draggable-dialog").should("not.be.visible");
+							});
+					});
+			});
+	});
+
+	it("draggable - keyboard support", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:535]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 10]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+										// Close dialog
+										cy.get("#draggable-close").click();
+										cy.get("#draggable-dialog").should("not.be.visible");
+									});
+							});
+					});
+			});
+	});
+
+	it("resizable - mouse support", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:606]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 11]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+								// Close dialog
+								cy.get("#resizable-close").click();
+								cy.get("#resizable-dialog").should("not.be.visible");
+							});
+					});
+			});
+	});
+
+	it("resizable - keyboard support", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:670]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 12]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+	it("Test initial focus when content is provided before the header", () => {
+		cy.mount(
+			<>
+				<Button id="openDialogFocusButton" onClick={() => {
+					const dialog = document.getElementById("dialog-focus");
+					if (dialog) {
+						(dialog as Dialog).open = true;
+					}
+				}}>Open Dialog</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: this button is obsolete, directly set the open attribute of the dialog

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 13]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+				<Dialog id="dialog-focus">
+					<Button id="fistButtonInDialog">First Button</Button>
+					<Button id="closeDialogFocusButton" onClick={() => {
+						const dialog = document.getElementById("dialog-focus");
+						if (dialog) {
+							(dialog as Dialog).open = false;
+						}
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: this handler is obsolete

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 14]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		);
+
+		// Open dialog
+		cy.get("#openDialogFocusButton").click();
+
+		// Assert initial focus is on the first button in the dialog
+		cy.get("#fistButtonInDialog").should("be.focused");
+	});
+
+	it("Test focus circularity", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:835]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 15]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		// Shift+Tab to the second button again
+		cy.realPress(["Shift", "Tab"]);
+		cy.get("#active-btn-2").should("be.focused");
+
+		// Close dialog
+		cy.realPress("Escape");
+		cy.get("#focus-circ").should("not.be.visible");
+	});
+
+	it("initial focus with autofocus", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:877]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 16]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+					<div slot="footer" class="dialogFooter">
+						<Button>Close</Button>
+					</div>
+				</Dialog>
+			</>
+		);
+
+		// Remove accessible-name attribute
+		cy.get("#dialog").invoke("removeAttr", "accessible-name");
+
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: delete this

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 17]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+							(dialog as Dialog).open = false;
+						}
+					}}>Close</Button>
+				</Dialog>
+				<Button id="draggable-open" onClick={() => {
+					const dialog = document.getElementById("draggable-dialog");
+					if (dialog) {
+						(dialog as Dialog).open = true;
+					}
+				}}>Open Draggable Dialog</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: remove this button

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 18]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		cy.get("#draggable-dialog")
+			.shadow()
+			.find(".ui5-popup-root")
+			.should("have.attr", "aria-label", "Draggable");
+
+		cy.get("#draggable-dialog")
+			.shadow()
+			.find(".ui5-popup-header-root")
+			.should("have.attr", "aria-describedby")
+			.then($el => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: unneeded `then` usage

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 19]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+			});
+
+		cy.get("#draggable-dialog")
+			.shadow()
+			.find(".ui5-popup-header-root")
+			.should("have.attr", "aria-roledescription", "Interactive Header");
+
+		// Close dialog
+		cy.get("#draggable-close").click();
+		cy.get("#draggable-dialog").should("not.be.visible");
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: delete this

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 20]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+	});
+
+	it("tests aria-describedby for default header", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:979]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 21]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+			.shadow()
+			.find(".ui5-popup-header-root")
+			.should("have.attr", "aria-roledescription", "Interactive Header");
+
+		// Close dialog
+		cy.get("#resizable-close").click();
+		cy.get("#resizable-dialog").should("not.be.visible");
+	});
+
+	it("tests aria-describedby for slot header", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1017]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 22]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+describe("Page scrolling", () => {
+	it("tests that page scrolling is blocked and restored", () => {
+		cy.mount(
+			<>
+				<Button id="btnOpenDialog" onClick={() => {
+					const dialog = document.getElementById("scroll-dialog");
+					if (dialog) {
+						(dialog as Dialog).open = true;
+					}
+				}}>Open Dialog</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: remove this button

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 23]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+				<Dialog id="scroll-dialog">
+					<Button id="btnCloseDialog" onClick={() => {
+						const dialog = document.getElementById("scroll-dialog");
+						if (dialog) {
+							(dialog as Dialog).open = false;
+						}
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: obsolete handler, directly use `cy.get("#scroll-dialog").invoke("attr", "open", ...);`

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 24]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+		// Close dialog
+		cy.get("#btnCloseDialog").click();
+
+		// Assert page scrolling is restored
+		cy.document().then(doc => {
+			const pageOverflow = window.getComputedStyle(doc.documentElement).overflow;
+			expect(pageOverflow).to.equal("visible");
+		});
+	});
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: `show` method no longer exists, so just delete this test

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 25]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+			// Assert scroll height is restored
+			cy.document().then(docAfterEscape => {
+				expect(docAfterEscape.documentElement.scrollHeight).to.equal(scrollHeightBefore);
+			});
+
+			// Disable scrolling
+			cy.get("#cbScrollable").click();
+		});
+	});
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: There seems to be no scroll height change before and after opening the dialog?

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 26]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+						if (confirmDialog) {
+							(confirmDialog as Dialog).open = true;
+						}
+					}
+
+					if (!preventClosing) {
+						preventClosing = true;
+					}
+				});
+			});
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: seems obsolete

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 27]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+							(confirmDialog as Dialog).open = false;
+						}
+					}}>Close</Button>
+				</Dialog>
+				<Button id="prevent" onClick={() => {
+					const dialog1 = document.getElementById("dialog1");
+					if (dialog1) {
+						(dialog1 as Dialog).open = true;
+					}
+				}}>Open Multiple Dialogs</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: This is misleading. It says multiple, but it opens only one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 28]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+	it("tests responsive paddings", () => {
+		cy.mount(
+			<>
+				<Button id="btnOpenDialog" onClick={() => {
+					const dialog = document.getElementById("dialog");
+					if (dialog) {
+						(dialog as Dialog).open = true;
+					}
+				}}>Open Dialog</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: obsolete

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 29]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+					<div slot="header">Dialog Header</div>
+					<div slot="footer">
+						<Button id="footerButton">Footer Button</Button>
+					</div>
+					<Button id="btnCloseDialog" onClick={() => {
+						const dialog = document.getElementById("dialog");
+						if (dialog) {
+							(dialog as Dialog).open = false;
+						}
+					}}>Close</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: obsolete

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 30]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+
+		const expectedPadding = "16px";
+		cy.get("#dialog").shadow().find(".ui5-popup-content").should("have.css", "padding-left", expectedPadding);
+		cy.get("#dialog").shadow().find(".ui5-popup-header-root").should("have.css", "padding-left", expectedPadding);
+		cy.get("#dialog").shadow().find(".ui5-popup-footer-root").should("have.css", "padding-left", expectedPadding);
+
+		cy.get("#btnCloseDialog").click();
+	});
+
+	it("tests removing of responsive paddings for the content", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1295]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 31]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		const expectedPadding = "16px";
+		const expectedContentPadding = "0px";
+		cy.get("#dialogNoPaddings").shadow().find(".ui5-popup-content").should("have.css", "padding-left", expectedContentPadding);
+		cy.get("#dialogNoPaddings").shadow().find(".ui5-popup-header-root").should("have.css", "padding-left", expectedPadding);
+		cy.get("#dialogNoPaddings").shadow().find(".ui5-popup-footer-root").should("have.css", "padding-left", expectedPadding);
+
+		cy.get("#btnCloseDialogNoPaddings").click();
+	});
+
+	it("tests media-range", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1317]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 32]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+			</>
+		);
+
+		cy.get("#btnOpenDialog").click();
+		cy.get("#dialog").should("have.attr", "media-range", "S");
+		cy.get("#btnCloseDialog").click();
+	});
+});
+
+describe("Dialog States", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1335]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 33]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		cy.get("#btn-error-state").click();
+
+		// Assert role
+		cy.get("#dialog-error-state")
+			.shadow()
+			.find(".ui5-popup-root")
+			.should("have.attr", "role", "alertdialog");
+
+		// Close dialog
+		cy.realPress("Escape");
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: obsolete

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 34]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+	});
+
+	it("tests information state dialog accessibility role", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1353]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 35]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		cy.get("#dialog-info-state")
+			.shadow()
+			.find(".ui5-popup-root")
+			.should("have.attr", "role", "dialog");
+
+		// Close dialog
+		cy.realPress("Escape");
+	});
+
+	it("tests success state dialog accessibility role", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1370]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 36]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		cy.get("#dialog-success-state")
+			.shadow()
+			.find(".ui5-popup-root")
+			.should("have.attr", "role", "dialog");
+
+		// Close dialog
+		cy.realPress("Escape");
+	});
+
+	it("tests warning state dialog accessibility role", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1390]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 37]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+	it("test dialog overlay when dialog isn't open", () => {
+		cy.mount(
+			<>
+				<Dialog id="dialog">
+					<Button id="btnCloseDialog" onClick={() => {
+						const dialog = document.getElementById("dialog");
+						if (dialog) {
+							(dialog as Dialog).open = false;
+						}
+					}}>Close</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: obsolete

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 38]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		);
+
+		// Assert block layer is hidden when dialog isn't open
+		cy.get("#dialog")
+			.shadow()
+			.find(".ui5-block-layer")
+			.should("not.have.class", "ui5-block-layer--visible");
+	});
+
+	it("test dialog overlay when dialog is open", () => {
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1425]: Comments from the previous test are also applicable to this one

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 39]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+	it("test :popover-open selector", () => {
+		cy.mount(
+			<>
+				<Dialog id="dialogOpen" open>
+					<Button id="btnCloseDialog" onClick={() => {
+						const dialog = document.getElementById("dialogOpen");
+						if (dialog) {
+							(dialog as Dialog).open = false;
+						}
+					}}>Close</Button>
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: obsolete

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 40]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+				</Dialog>
+			</>
+		);
+
+		// Assert dialog matches :popover-open selector
+		cy.get("#dialogOpen").should("have.attr", "open");
```
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx]: you should make assertion for `:popover-open` CSS selector

File: packages/main/cypress/specs/Dialog.cy.tsx
Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1]: `expect` calls should happen in `should` callbacks, not `then` to leverage cypress's polling

Comment (dimovpetar) [inline packages/main/cypress/specs/Dialog.cy.tsx:1]: There are some warnings in the console for attributes usages. Replace such with properties

**Review by dimovpetar — CHANGES_REQUESTED** [review]

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 41]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
 			.should("not.be.visible");
 	});
 });
+
+describe("Dialog general interaction", () => {
+	it("tests dialog toggling", () => {
+
+		const closeDialog = () => {
+			const dialog = document.getElementById("dialog");
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Dialog.cy.tsx]: use [custom-tag] to select element when they are one of a type.

when configuring custom element use camelCase to set properties insetad of attributes. Keep examples simple as possible with everything used for the test only

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 42]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+						if (dialog) {
+							(dialog as Dialog).open = false;
+						}
+					}}>Close with Method</Button>
+					<Button id="btnCloseWithAttr">Close with Attribute</Button>
+				</Dialog>
+			</>
+		);
+
+		cy.get("#dlgAttr")
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Dialog.cy.tsx]: both of these tests are the same. this is a framework logic and should not be tested if there are no some edge cases.

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 43]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+				cy.viewport(2000, 2000);
+
+				cy.get("#dialog").then(function (dialog) {
+					dialog[0].setAttribute("open", "true")
+				})
+
+				// Capture position after resize
+				cy.get("#dialog")
+					.then(dialogAfterResize => {
+						const topAfterScreenResize = parseInt(dialogAfterResize.css("top"));
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Dialog.cy.tsx:415]: should instead of then when using expect inside the callback

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 44]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+		// Close dialog
+		cy.get("#dialog-close").click();
+		cy.get("#dialog").should("not.be.visible");
+	});
+
+	it("draggable and resizable dialog repositions after screen resize", () => {
+		const dragDialog = (dialogSelector: string, x: number, y: number) => {
+			cy.get(dialogSelector)
+				.find("#header-slot")
+				.trigger("mousedown", { which: 1 })
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Dialog.cy.tsx]: real events

File: packages/main/cypress/specs/Dialog.cy.tsx [hunk 45]
```diff
@@ -280,3 +286,1538 @@ describe("Events", () => {
...
+					<Button id="draggable-and-resizable-close" onClick={closeDialog}>Close</Button>
+				</Dialog>
+			</>
+		);
+
+		cy.viewport(1000, 800);
+
+		// Open dialog
+		cy.get("#draggable-and-resizable-dialog").then(function (dialog) {
+			dialog[0].setAttribute("open", "true")
```
Comment (nnaydenow) [inline packages/main/cypress/specs/Dialog.cy.tsx]: invoke instead then with setAttribute

File: packages/main/cypress/specs/Dialog.cy.tsx (hunk 40 above)
Comment (nnaydenow) [reply packages/main/cypress/specs/Dialog.cy.tsx]: there is custom command which could be used in all of the tests, not only here
